### PR TITLE
Adding extra options for MOABB dataset

### PR DESCRIPTION
**Pull request by bruAristimunha**
Hi @PierreGtch!

In this PR, I am doing:

1) Adding an option to pass a moabb object directly to braindecode moabb dataset.
2) Adding an extra flag to use the cache dir and pre-procesing option to the moabb dataset.

Can you review it?

### CODE REVIEW

**Comment by bruAristimunha** [comment]
FYI: @gustavohenriquesr 

Can you test this code and check if it is working in our case until the next meeting?

Thanksss!

**Comment by bruAristimunha** [comment]
The error on CI is not related; https://www.physionet.org/ is down.

File: braindecode/datasets/moabb.py [hunk 1]
```diff
@@ -62,28 +66,44 @@ def _annotations_from_moabb_stim_channel(raw, dataset):
...
+    dataset_name: str | moabb.datasets.base.BaseDataset
         the name of a dataset included in moabb
     subject_ids: list(int) | int
         (list of) int of subject(s) to be fetched
     dataset_kwargs: dict, optional
         optional dictionary containing keyword arguments
         to pass to the moabb dataset when instantiating it.
+    data_load_kwargs: dict, optional
+        optional dictionary containing keyword arguments
+        to pass to the moabb dataset when loading the data.
```
Comment (PierreGtch) [inline braindecode/datasets/moabb.py:86]:
```suggestion
        to pass to the moabb dataset's load_data method.
```

File: braindecode/datasets/moabb.py [hunk 2]
```diff
@@ -99,11 +119,21 @@ class MOABBDataset(BaseConcatDataset):
     dataset_kwargs: dict, optional
         optional dictionary containing keyword arguments
         to pass to the moabb dataset when instantiating it.
+    dataset_load_kwargs: dict, optional
+        optional dictionary containing keyword arguments
+        to pass to the moabb dataset when loading the data.
```
Comment (PierreGtch) [inline braindecode/datasets/moabb.py:124]:
```suggestion
        to pass to the moabb dataset's load_data method.
```

**Comment by PierreGtch** [comment]
Just small doc comments. Otherwise, cool feature :)

**Pull request merged**